Parse int64 sizes/strides

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64077
* __->__ #64076

We were parsing sizes into int32s, so if you had a tensor with more
than 2^32 elements, you couldn't represent it.

Differential Revision: [D30521116](https://our.internmc.facebook.com/intern/diff/D30521116/)

bertmaher committed:
Parse int64 sizes/strides

We were parsing sizes into int32s, so if you had a tensor with more
than 2^32 elements, you couldn't represent it.

Differential Revision: [D30521116](https://our.internmc.facebook.com/intern/diff/D30521116/)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good, but maybe add a test to https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_irparser.cpp ?

bertmaher committed:
Update on "Parse int64 sizes/strides"


We were parsing sizes into int32s, so if you had a tensor with more
than 2^32 elements, you couldn't represent it.

Differential Revision: [D30521116](https://our.internmc.facebook.com/intern/diff/D30521116/)

[ghstack-poisoned]

bertmaher commented:
@bertmaher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30521116).

bertmaher commented:
@bertmaher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30521116).

bertmaher committed:
Update on "Parse int64 sizes/strides"


We were parsing sizes into int32s, so if you had a tensor with more
than 2^32 elements, you couldn't represent it.

Differential Revision: [D30521116](https://our.internmc.facebook.com/intern/diff/D30521116/)

[ghstack-poisoned]
